### ISSUE TITLE
[FEA] Full coverage of datetime methods in cudf-polars

### ISSUE THREAD

**Issue by wence-** [rapidsai/cudf#16481]
We have already implemented a subset of datetime methods in cudf polars, based on prioritisation from initial benchmark needs.

We should build out the support to cover all the methods polars supports (in the [polars.Expr.dt](https://docs.pola.rs/api/python/stable/reference/expressions/temporal.html) namespace).

From a cursory glance it looks like most of these are covered by existing libcudf functionality. Where they are not, this can provide some indication of what additional features we might wish for in libcudf datetime processing routines.

Some existing related feature requests:

- [ ] https://github.com/rapidsai/cudf/issues/16226
- [ ] https://github.com/rapidsai/cudf/issues/16174
- [ ] https://github.com/rapidsai/cudf/issues/16196

**Comment by jberg7** [comment]
Hi team, thanks for your work here! Does this ticket cover timezone support, or is there something more specific I can track?
Timezone support is a big blocker for using the polars gpu engine on timeseries data, which is often timezone-aware.

**Comment by btepera** [comment]
@jberg7 Yes, this issue is meant to capture support for timezones as well. How are you using timezones today, mainly through `convert_time_zone` and `replace_time_zone`? Would be interested in learning more about your specific use case if you're open to discussing.

**Comment by jberg7** [comment]
@btepera thanks for the response! Yes that's basically my use case, a contrived example would be something like:
```python
import polars as pl
import datetime

lf = pl.LazyFrame({"utcnow": [datetime.datetime.now(tz=datetime.timezone.utc)]}).with_columns(pl.col("utcnow").dt.replace_time_zone(None).alias("now"))
lf.collect(engine=pl.GPUEngine(raise_on_fail=True))

# polars.exceptions.ComputeError: 'cuda' conversion failed: NotImplementedError: ('Query execution with GPU not possible: unsupported operations.\nThe errors were:\n- NotImplementedError: Time zone support', [NotImplementedError('Time zone support')])
```

**Comment by mroeschke** [comment]
The remaining, unsupported datetime expressions are captured in https://github.com/rapidsai/cudf/issues/23151 so closing in favor of that issue
